pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,294 @@
+        get_process=lambda v: SystemStatusCode(int(v, base=16)),
+    )
+
+    local_lockout = Instrument.setting(
+        "%s",
+        """Set the local interface lockout.
+
+        :type: bool
+            ``True``: disables the local interface
+            ``False``: enables it.
+
+        """,
+        validator=strict_discrete_set,
+        values={True: "*LOCK", False: "*UNLOCK"},
+        map_values=True
+    )
+
+    save_settings = Instrument.setting(
+        "*SAV %d",
+        """Save the current configuration in non-volatile memory.
+
+        :type: int
+        """,
+        validator=strict_range,
+        values=[1, 5]
+    )
+
+    recall_settings = Instrument.setting(
+        "*RCL %d",
+        """Recall the saved configuration from non-volatile memory.
+
+        :type: int
+        """,
+        validator=strict_range,
+        values=[1, 5]
+    )
+
+    selected_channel = Instrument.control(
+        "INST?", "INST %s",
+        """Control the selected channel of the instrument.
+
+        :type: int
+
+        """,
+        validator=strict_discrete_set,
+        values={1: "CH1"},  # This dynamic property should be updated for multi-channel instruments
+        map_values=True,
+        dynamic=True
+    )
+
+    set_4W_mode = Instrument.setting(
+        "MODE:SET %s",
+        """Configure 4-wire mode.
+
+        :type: bool
+            ``True``: enables 4-wire mode
+            ``False``: disables it.
+
+        """,
+        validator=strict_discrete_set,
+        values={False: "2W", True: "4W"},
+        map_values=True
+    )
+
+    def save_config(self, index):
+        """Save the current config to memory.
+
+        :param index:
+            string: index of the location to save the configuration
+        """
+        self.save_settings = index
+
+    def recall_config(self, index):
+        """Recall a config from memory.
+
+        :param index:
+            string: index of the location from which to recall the configuration
+        """
+        self.recall_settings = index
+
+
+class SPDChannel(object):
+    """ The channel class for Siglent SPDxxxxX instruments.
+    """
+
+    def __init__(self, instrument, channel: int = 1):
+        self.instrument = instrument
+        self.channel = channel
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    current = Instrument.measurement(
+        "MEAS:CURR? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    power = Instrument.measurement(
+        "MEAS:POWE? CH{channel}",
+        """Measure the channel output voltage.
+
+        :type: float
+        """
+    )
+
+    set_current = Instrument.control(
+        "CH{channel}:CURR?", "CH{channel}:CURR %g",
+        """Control the output current configuration of the channel.
+
+        :type: float
+
+        """,
```

Comment:
For some reason, when I try and and build the docs without this space I end up with:
`C:\Users\dschmeer\Projects\pymeasure\pymeasure\instruments\siglenttechnologies\siglent_spdbase.py:docstring of pymeasure.instruments.siglenttechnologies.siglent_spdbase.SPDChannel:1: WARNING: Field list ends without a blank line; unexpected unindent.`
It did this in a few places in SPDBase, is there a known reason for this? If not Ill just leave the newline in

Reply:
I've done a bit more testing, it seems like there is a sphinx whitespace bug going on here. I have two options which work, one is the one included above. The other option is to replace the line in the docstring as:
`  :type : float
   """,`. Personally my preference will be to keep the newline because it looks tidier

Reply:
Hm, could be a Sphinx/ReST quirk or bug (couldn't find a report in the sphinx repo with a quick search), resolve it as you want. Great that you checked doc generation locally!

Reply:
or did that work out in the end?